src/functional/ffront/symbol_makers.py

```diff
@@ -147,31 +147,35 @@ def make_symbol_type_from_typing(
             }
 
             # TODO(tehrengruber): print better error when no return type annotation is given
-            return common_types.FunctionType(
+            return ct.FunctionType(
                 args=args, kwargs=kwargs, returns=recursive_make_symbol(return_type)
             )
-        case runtime.Offset:
-            return common_types.OffsetType()
+
+        case type() as t if issubclass(t, common.Dimension):
+            return ct.OffsetType()
 
     raise TypingError(f"'{type_hint}' type is not supported")
 
 
-def make_symbol_type_from_value(value: Any) -> common_types.SymbolType:
-    """Make a symbol type from a Python value."""
-    if isinstance(value, type) or type(value).__module__ == "typing":
-        # we don't have types of types so disallow this
-        raise ValueError("The type of a symbol can not be a type itself.")
-    # TODO(tehrengruber): What we expect here currently is a GTCallable. Maybe
-    #  we should check for the protocol in the future?
-    if hasattr(value, "__gt_type__"):
-        symbol_type = value.__gt_type__()
+def make_symbol_type_from_value(value: Any) -> ct.SymbolType:
+    """Make a symbol node from a Python value."""
+    if isinstance(value, ct.SymbolType):
+        # Temporary hack for type deductions with shifted fields
+        symbol_type = value
+    elif isinstance(value, runtime.Offset):
+        # Temporary hack for type deductions with shifted fields
+        symbol_type = ct.OffsetType(source=value.source, target=value.target)
```

Comment:
This other workaround can also be handled using the  `__gt_type__` attribute. Most likely it is only needed to add this line to the `runtime.Offset` class:
```python
class Offset:
  ...

    def __gt_type__(self):
        return ct.OffsetType(source=self.source, target=self.target)
```
I would also add a TODO comment mentioning that a cleaner solution will be to use the encode the `OffsetType` as an actual type also valid for the iterator prototype

Reply:
Actually, it would probably be better to eventually lift both `runtime.Offset` and `common_types` into `functional`.